File: sdks/dart/lib/src/regions.dart
import 'package:dio/dio.dart';

class Regions {
  final Dio _dio;

  Regions(this._dio);

  Future<Map<String, dynamic>> list() async {
    final res = await _dio.get('/v1/regions');
    return res.data;
  }

  Future<Map<String, dynamic>> get(String regionId) async {
    final res = await _dio.get('/v1/regions/$regionId');
    return res.data;
  }

  Future<Map<String, dynamic>> getActive() async {
    final res = await _dio.get('/v1/regions/active');
    return res.data;
  }

  Future<Map<String, dynamic>> setActive(String regionId) async {
    final res = await _dio.put('/v1/regions/active', data: {
      'regionId': regionId,
    });
    return res.data;
  }

  Future<Map<String, dynamic>> getHealth(String regionId) async {
    final res = await _dio.get('/v1/regions/$regionId/health');
    return res.data;
  }
}

File: sdks/js/src/regions.ts
import type { Applad } from './client';

export class Regions {
  constructor(private client: Applad) {}

  /** List available regions. */
  list() {
    return this.client.call('GET', '/regions');
  }

  /** Get a specific region. */
  get(regionId: string) {
    return this.client.call('GET', `/regions/${regionId}`);
  }

  /** Get the active region for the current project. */
  getActive() {
    return this.client.call('GET', '/regions/active');
  }

  /** Set the active region for the current project. */
  setActive(regionId: string) {
    return this.client.call('PUT', '/regions/active', { regionId });
  }

  /** Get region health / latency info. */
  getHealth(regionId: string) {
    return this.client.call('GET', `/regions/${regionId}/health`);
  }
}
